packages/docusaurus-plugin-content-docs/src/tags.ts

```diff
@@ -25,3 +31,47 @@ export function getVersionTags(docs: DocMetadata[]): VersionTags {
     };
   });
 }
+
+export const tagDefinitionSchema = Joi.object().pattern(
+  Joi.string(),
+  Joi.object({
+    label: Joi.string().required(),
+    description: Joi.string().required(),
+  }),
+);
+
+export function validateDefinedTags(tags: unknown): Joi.ValidationResult {
+  return tagDefinitionSchema.validate(tags);
+}
+
+export function createTagSchema(tags: string[]): Joi.Schema {
+  return Joi.array().items(Joi.string().valid(...tags));
+}
+
+export function validateFrontMatterTags({
+  frontMatterTags,
+  validTagsSchema,
+  source,
+  onBrokenTags,
+}: {
+  frontMatterTags: FrontMatterTag[] | undefined;
+  validTagsSchema: Joi.Schema<string[]>;
+  source: string;
+  onBrokenTags: MetadataOptions['onBrokenTags'];
+}): void {
+  if (frontMatterTags === undefined || !Array.isArray(frontMatterTags)) {
+    return;
+  }
+
+  const labels = frontMatterTags.map((tag) =>
+    typeof tag === 'string' ? tag : tag.permalink,
+  );
+
+  const tagList = validTagsSchema.validate(labels);
+
+  if (tagList.error) {
+    logger.report(onBrokenTags)(
+      `Broken tags found in ${source} [${labels}] : ${tagList.error.message}`,
+    );
+  }
```

Comment:
No need to use Joi for that

Collect inline tags, and if `onUnknownTags` emit custom error message

Maybe we should consider naming the option `onInlineTags` ?